### PR TITLE
Sheet: Component Contribution

### PR DESCRIPTION
**Pull request by michael-iden**
**This change:** (check at least one)

- [X] Adds a new feature
- [ ] Fixes a bug
- [ ] Improves maintainability
- [ ] Improves documentation
- [ ] Is a release activity

**Is this a breaking change?** (check one)

- [ ] Yes
- [X] No

**Is the:** (complete all)

- [x] Title of this pull request clear, concise, and indicative of the issue number it addresses, if any?
- [ ] Test suite(s) passing?
- [ ] Code coverage maximal?
- [ ] Changeset added?
- [ ] Component status page up to date?

**What does this change address?**
Adds a sheet component to Pharos

**How does this change work?**
A modification of the mobile modal to show a sheet on any screen size

**Additional context**

### CODE REVIEW

File: packages/pharos/src/components/sheet/pharos-sheet.ts [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+
+      if (this.open) {
+        body?.classList.add('pharos-sheet__body');
+        this._focusContents();
+      } else {
+        body?.classList.remove('pharos-sheet__body');
+        this._returnTriggerFocus();
+      }
+
+      if (changedProperties.get('open') !== undefined) {
```
Comment (chrisjbrown) [inline packages/pharos/src/components/sheet/pharos-sheet.ts:128]: is this condition not covered by the upper closure?
https://github.com/ithaka/pharos/pull/603/files#diff-1f2222a8dc07b2a9918b11299ed4578fa635f936cb9b05ff423597cc0ef8d3e1R117

File: packages/pharos/src/components/sheet/pharos-sheet.ts [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+   */
+  @property({ type: String, reflect: true })
+  public header = 'Sheet header';
+
+  /**
+   * Indicates if the sheet footer should contain a divider.
+   * @attr footer-divider
+   */
+  @property({ type: Boolean, reflect: true, attribute: 'footer-divider' })
+  public footerDivider = false;
```
Comment (chrisjbrown) [inline packages/pharos/src/components/sheet/pharos-sheet.ts:79]: is this used?

Comment (michael-iden) [reply packages/pharos/src/components/sheet/pharos-sheet.ts:79]: Probably not. I think this was leftover from copy and pasting the modal component as a starting point

Comment (jialin-he) [reply packages/pharos/src/components/sheet/pharos-sheet.ts:79]: Yeah, it is used in css only, to determine if we want to show a border on the footer

File: packages/pharos/src/components/sheet/pharos-sheet.ts (hunk 1 above)
Comment (jialin-he) [reply packages/pharos/src/components/sheet/pharos-sheet.ts:128]: @michael-iden do you know why do we add the additional checking here?

Comment (daneah) [reply packages/pharos/src/components/sheet/pharos-sheet.ts:128]: We should ensure only code we need is being used, and a fundamental understanding of what's going on with each bit would be a good place to start. Lit's [Reactive properties](https://lit.dev/docs/components/properties/) is a good reference.

Comment (jialin-he) [reply packages/pharos/src/components/sheet/pharos-sheet.ts:128]: Ah so from what I see in the doc, "changedProperties is a Map with keys that are the names of changed properties and values that are the corresponding previous values". So `changedProperties.has('open')` just check if the map has key `open` but `changedProperties.get('open')` checks the actual value of `open`?

File: packages/pharos/src/components/sheet/pharos-sheet.ts [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+   */
+  @property({ type: Boolean, reflect: true })
+  public expanded = false;
+
+  /**
+   * Indicates if the sheet is allowed to expand.
+   * @attr enbaleExpansion
+   */
+  @property({ type: Boolean, reflect: true })
+  public enbaleExpansion = true;
```
Comment (daneah) [inline packages/pharos/src/components/sheet/pharos-sheet.ts:58]: Misspelled word, here.

File: packages/pharos/src/components/sheet/pharos-sheet.scss [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  height: 50%;
+  width: 100%;
+  pointer-events: auto;
+  overflow-y: scroll;
+  background-color: var(--pharos-modal-color-background-content);
+  background-clip: padding-box;
+  border: 1px solid var(--pharos-modal-color-border-content);
```
[the remark this answers is not in the document]
Comment (jialin-he) [reply packages/pharos/src/components/sheet/pharos-sheet.scss]: Updated!

File: packages/pharos/src/components/sheet/pharos-sheet.ts (hunk 3 above)
Comment (jialin-he) [reply packages/pharos/src/components/sheet/pharos-sheet.ts]: ahhhh good eye!

**Review by chrisjbrown — APPROVED** [review]

**Review by satya-achanta-venkata — APPROVED** [review]

File: packages/pharos/src/components/sheet/pharos-sheet.ts [hunk 5]
```diff
@@ -0,0 +1,367 @@
...
+  protected override updated(changedProperties: PropertyValues): void {
+    if (changedProperties.has('open')) {
+      const body = document.querySelector('body');
+
+      if (this.open) {
+        body?.classList.add('pharos-sheet__body');
+        this._focusContents();
+      } else {
+        body?.classList.remove('pharos-sheet__body');
+        this._returnTriggerFocus();
```
Comment (daneah) [inline packages/pharos/src/components/sheet/pharos-sheet.ts:125]: Is there a reason we use the `pharos-` prefix here?

Comment (jialin-he) [reply packages/pharos/src/components/sheet/pharos-sheet.ts:125]: actually I don't think it is used. deleted.

**Pull request merged**